### PR TITLE
Fix find_children not searching recursively in some cases

### LINKED ISSUE #63757: `find_children` returns `[]` if match string is given
### Godot version

v4.0.alpha13.official [59dcddc45]

### System information

Windows 11, Nvidia GeForce GTX 1050, 27.21.14.5771

### Issue description

`find_children` method of `Node` returns empty list even if it is looking for a node with a name is present in the tree! 
It finds the node if the `match_string` is `"*"`.

### Steps to reproduce
```
find_children("*name*", "", true, true)  # returns []
find_children("*") # returns [nodes! whoohoo!]
```

### Minimal reproduction project
[MRP.zip](https://github.com/godotengine/godot/files/9231350/MRP.zip)

### PR DESCRIPTION
**Pull request by heppocogne**
This commit fixes #63757

### CODE REVIEW

**Comment by YuriSizov** [comment]
The fix itself makes sense, but we don't normally resort to `goto`...

**Comment by heppocogne** [comment]
Honestly, I don't want to use `goto` either. So, should I replace `goto` with the following code?
```C++
if (p_recursive) {
	ret.append_array(cptr[i]->find_children(p_pattern, p_type, true, p_owned));
}
continue;
```

**Comment by Maran23** [comment]
> Honestly, I don't want to use `goto` either. So, should I replace `goto` with the following code?

Yes, or think about another way.

File: scene/main/node.cpp [hunk 1]
```diff
@@ -1417,12 +1417,12 @@ TypedArray<Node> Node::find_children(const String &p_pattern, const String &p_ty
 	int ccount = data.children.size();
 	for (int i = 0; i < ccount; i++) {
 		if (p_owned && !cptr[i]->data.owner) {
-			continue;
+			goto next_loop;
 		}
```
Comment (kleonc) [inline scene/main/node.cpp:1421]: Regarding this change I think the previous behavior might be intended, see https://github.com/godotengine/godot/pull/75459#issuecomment-1511832318.

**Comment by akien-mga** [comment]
Superseded by #75459.

**Pull request closed without merging**